File: src/main/java/com/yaskovdev/motivated/stay/UserController.java
package com.yaskovdev.motivated.stay;

import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

import static org.springframework.http.ResponseEntity.notFound;
import static org.springframework.http.ResponseEntity.ok;

@RestController
@RequestMapping("/users")
@RequiredArgsConstructor
class UserController {

    private final UserRepository repository;

    @PutMapping("/{id}")
    ResponseEntity<User> create(@PathVariable("id") final String id, @RequestBody final User user) {
        user.setId(id);
        return repository.existsById(id) ? ok(user) : ok(repository.insert(user));
    }

    @GetMapping("/{id}")
    ResponseEntity<User> read(@PathVariable("id") final String id) {
        return repository.findById(id).map(ResponseEntity::ok).orElse(notFound().build());
    }

    @GetMapping
    ResponseEntity<List<User>> readAll() {
        return ok(repository.findAll());
    }

    @PostMapping("/{id}/deletion")
    ResponseEntity<User> delete(@PathVariable("id") final String id) {
        repository.deleteById(id);
        return ok().build();
    }
}

File: src/main/java/org/motivepick/web/UserController.kt
package org.motivepick.web

import org.motivepick.domain.entity.UserEntity
import org.motivepick.service.UserService
import org.springframework.http.ResponseEntity
import org.springframework.http.ResponseEntity.notFound
import org.springframework.http.ResponseEntity.ok
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
internal class UserController(private val userService: UserService) {

    @GetMapping("/user")
    fun read(): ResponseEntity<UserEntity> = userService.readCurrentUser()?.let { ok(it) } ?: notFound().build()
}
